dev/Repeater/FlowLayoutAlgorithm.cpp

```diff
@@ -601,12 +601,14 @@ void FlowLayoutAlgorithm::PerformLineAlignment(
     int countInLine,
     float spaceAtLineStart,
     float spaceAtLineEnd,
+    float lineSize,
     FlowLayoutAlgorithm::LineAlignment lineAlignment,
     const wstring_view& layoutId)
 {
     for (int rangeIndex = lineStartIndex; rangeIndex < lineStartIndex + countInLine; ++rangeIndex)
     {
         auto bounds = m_elementManager.GetLayoutBoundsForRealizedIndex(rangeIndex);
+        bounds.*MajorSize() = lineSize;
```

Comment:
What does this... do?
(still feeling really new to C++/Cx)

Reply:
I have a lot of trouble wrapping my head around this every time it comes up because I think there's parentheses that would really help clarify. But I *believe* what's happening here is that the MajorSize function is a member function of the current class (FlowLayoutAlgorithm, by way of OrientationBasedMeasures). MajorSize function returns a *pointer to a member* of the `winrt::Rect` class (hence the return of MajorSize is `winrt::Rect::*` and return the address of one of the winrt::Rect methods). Then MajorSize is evaluated off of the bounds variable ... which is a winrt::Rect.

I struggle because I can find lots of documentation about the .* (pointer to member) operator, but no examples doing this specific strategy. I don't know why, but this particular operator usage seems to be my kryptonite.

Reply:
Okay, I think I get it. Buuut…. if that had bee something like this:
```
SetMajorSize(bounds, lineSize)
```
I wouldn't have wondered. The pointer to a member thing is clever... but I think there are easier to understand options.

Reply:
You can basically think of it as a member pointer rather than a global pointer.  `winrt::Rect* rect` stores a value that says, "Starting from 0, you can find a winrt::Rect object at whatever memory position `rect` stores."  Conversely, `float winrt::Rect::* rectValue` stores a value that says, "Starting from the beginning of a winrt::Rect object, you can find a float value at whatever offset is stored in `rectValue`."

It's a weird feature that isn't unique to C++/WinRT; it exists in vanilla C++ too.  It basically just serves to replace an if/else statement that'd look like this:

```
float majorSize = 0;

if (majorDirection == Direction::Horizontal)
{
    majorSize = bounds.Width;
}
else
{
    majorSize = bounds.Height;
}
```

Reply:
I think if the syntax we wrote was:

```
bounds.*(MajorSize()) = lineSize;
```

Then at least the person reading it says "oh, ok, MajorSize is a function that's evaluated first". Maybe that would have helped me find it less confusing? But still I am worried the cognitive load for the benefit is low.

Can I suggest a compromise like this instead? It gets similar use-site syntax but doesn't use a rarely-seen language feature:
```
float& MajorMinor(winrt::Rect &rect)
{
    return isHorizontal ? rect.X : rect.Y;
}

// use site:
MajorMinor(bounds) = lineSize;
```

Reply:
Makes sense. I'm not a big fan of this syntax either - so I am happy to change this. I'll take this as a separate bug/work item since it is not related to this bug fix. Tracked by https://github.com/microsoft/microsoft-ui-xaml/issues/678